src/saveload/details.rs

```diff
@@ -0,0 +1,176 @@
+use std::error::Error;
+
+use serde::de::DeserializeOwned;
+use serde::ser::Serialize;
+
+use {Component, Entity, ReadStorage, SystemData, WriteStorage};
+
+use super::marker::Marker;
+use error::NoError;
+
+#[derive(Serialize, Deserialize)]
+#[serde(bound = "")]
+pub struct EntityData<M: Marker, E, T: Components<M::Identifier, E>> {
+    pub marker: M,
+    pub components: T::Data,
+}
+
+/// Trivial component is anything that is Component + Copy + DeserializeOwned + Serialize
+pub trait SerializableComponent: Component + DeserializeOwned + Serialize {}
```

Comment:
where do we actually need this? It looks like `SaveLoadComponent` could be implemented straight for `C: Component + DeserializeOwned + Serialize`

Reply:
Of course it could as this one is implemented for `C: Component + DeserializeOwned + Serialize`. It's just a handy bundle of traits to make bounds look nicer.